scripts/project.js

```diff
@@ -1,44 +1,53 @@
-function Project(opts){
-  this.title = opts.title;
-  this.projectUrl = opts.projectUrl;
-  this.publishedOn = opts.publishedOn;
-  this.projectDescription = opts.projectDescription;
-}
+(function(module){
+  function Project(opts){
+    this.title = opts.title;
+    this.projectUrl = opts.projectUrl;
+    this.publishedOn = opts.publishedOn;
+    this.projectDescription = opts.projectDescription;
+  }
 
-Project.all = [];
+  Project.all = [];
 
-Project.prototype.toHtml = function() {
-  var source = $('#project-template').html();
-  var template = Handlebars.compile(source);
-  return template(this);
-}
+  Project.prototype.toHtml = function() {
+    var source = $('#project-template').html();
+    var template = Handlebars.compile(source);
+    return template(this);
+  }
 
-Project.loadAll = function(rawData) {
-  rawData.forEach(function(ele) {
-    Project.all.push(new Project(ele));
-  })
-};
+//  Project.loadAll = function(rawData) {
+//    rawData.forEach(function(ele) {
+//      Project.all.push(new Project(ele));
+//    })
+//  };
 
-Project.fetchAll = function() {
-  var storedEtag;
-  $.ajax({
-    type: "HEAD",
-    url: "data/portfolioProjects.json",
-    success: function(data, message, xhr) {
-      storedEtag = xhr.getResponseHeader("etag");
-      if (storedEtag === localStorage.etag && localStorage.rawData) {
-        Project.loadAll(JSON.parse(localStorage.rawData));
-        projectView.initIndexPage();
-      } else {
-        $.getJSON('data/portfolioProjects.json', function(data) {
-          console.log(data);
-          Project.loadAll(data);
-          localStorage.setItem("etag", JSON.stringify(storedEtag));
-          localStorage.setItem('rawData', JSON.stringify(data));
+  Project.loadAll = function(rawData){
+    Project.all = rawData.map(function(ele) {
+      return new Project(ele);
+    });
+  };
+
+  Project.fetchAll = function() {
+    var storedEtag;
+    $.ajax({
+      type: "HEAD",
+      url: "data/portfolioProjects.json",
+      success: function(data, message, xhr) {
+        storedEtag = xhr.getResponseHeader("etag");
+        if (storedEtag === localStorage.etag && localStorage.rawData) {
+          Project.loadAll(JSON.parse(localStorage.rawData));
           projectView.initIndexPage();
-          });
+        } else {
+          $.getJSON('data/portfolioProjects.json', function(data) {
+            console.log(data);
+            Project.loadAll(data);
+            localStorage.setItem("etag", JSON.stringify(storedEtag));
+            localStorage.setItem('rawData', JSON.stringify(data));
+            projectView.initIndexPage();
+            });
+        }
```

Comment:
I'd either all use the same name, or pick some very semantic names for the data at each part of the process. It depends on which you want to emphasize: that this is all just different phases of the same data, or that this data is going through various transformations. Whichever is clearest for you.